### ISSUE TITLE
Fix GTK4 apps relying on gtk-update-icon-cache

### ISSUE THREAD

**Issue by jtojnar** [NixOS/nixpkgs#302605]
https://github.com/NixOS/nixpkgs/pull/301865 made it so that the GTK 3 dependency is not needed in Nixpkgs but it should really be fixed upstream so that we can clean up the expressions and other parties can also benefit. [Example of upstream MR](https://gitlab.gnome.org/GNOME/swell-foop/-/merge_requests/34)

The following apps are affected

- [ ] [`hydrapaper`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name/hy/hydrapaper/package.nix)
- [ ] [`passes`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name/pa/passes/package.nix)
- [ ] [`glide-media-player`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name/gl/glide-media-player/package.nix)
- [ ] [`smile`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/by-name/sm/smile/package.nix)
- [ ] [`confy`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/applications/misc/confy/default.nix)
- [ ] [`gnome-builder`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/applications/editors/gnome-builder/default.nix) ([`meson.post_install()` will not be sufficient](https://gitlab.gnome.org/GNOME/gnome-builder/-/merge_requests/609))
- [ ] [`wike`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/applications/misc/wike/default.nix)
- [ ] [`parlatype`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/applications/audio/parlatype/default.nix)
- [ ] [`gnome-frog`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/applications/misc/gnome-frog/default.nix)
- [ ] [`spot`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/applications/audio/spot/default.nix)
- [ ] [`eartag`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/applications/audio/eartag/default.nix)
- [ ] [`p2p/torrential`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/applications/networking/p2p/torrential/default.nix)
- [ ] [`blanket`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/applications/audio/blanket/default.nix)
- [ ] [`gnome-sound-recorder`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/desktops/gnome/apps/gnome-sound-recorder/default.nix)
- [ ] [`blackbox-terminal`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/applications/terminal-emulators/blackbox-terminal/default.nix)
- [ ] [`gabutdm`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/applications/networking/gabutdm/default.nix)
- [ ] [`giara`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/applications/networking/giara/default.nix)
- [x] [`video-trimmer`](https://github.com/NixOS/nixpkgs/tree/master/pkgs/applications/video/video-trimmer/default.nix) (https://gitlab.gnome.org/YaLTeR/video-trimmer/-/merge_requests/12)

**Comment by linsui** [comment]
I can't fork blackbox on gitlab.gnome.org, sorry. :)

**Comment by alexshpilkin** [comment]
Hm, sorry, I seem to be lacking context here. How do I reproduce / motivate this? Do things just not build if gtk3 is not installed (even outside of NixOS)?

**Comment by linsui** [comment]
@alexshpilkin It's fixed with this patch: https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/audio/parlatype/default.nix#L61. And it's better to fix it upstream with such a patch: https://gitlab.gnome.org/GNOME/swell-foop/-/merge_requests/34. The script expects the gtk3 is installed.

**Comment by alexshpilkin** [comment]
@linsui Right, I can ape the patch, I’m just trying to understand what’s being fixed (and going to need to explain that to the maintainer).

So if I’ve understood this correctly: there is `gtk-update-icon-cache` in Gtk 3 and `gtk4-update-icon-cache` in Gtk 4, and they maintain largely the same thing in compatible ways, so we want people to stop calling the Gtk 3 tool by name when they are building/installing a Gtk≠3 app (because pulling in Gtk 3 as a dependency is silly).

**Comment by jtojnar** [comment]
That is correct.

**Comment by koppor** [comment]
@alexshpilkin Thank you for the clarifications and the "example" PR.

@jtojnar PR for "smile" was merged (https://github.com/mijorus/smile/pull/79). IDK when there will be a new release.

**Comment by alexshpilkin** [comment]
@jtojnar The patch to `parlatype` (https://github.com/gkarsay/parlatype/pull/113) has been accepted. Upstream releases are usually infrequent—I guess we could ask for one if needed.
